### PR TITLE
Create a Docker base image with poetry

### PR DESCRIPTION
**Pull request by tczekajlo**
**Proposed changes**:
- Create a separate Docker base image with poetry

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docker/Dockerfile.full [hunk 1]
```diff
@@ -2,11 +2,17 @@
...
 ARG BASE_IMAGE_HASH
 ARG BASE_MITIE_IMAGE_HASH
+ARG POETRY_VERSION
 
 FROM ${IMAGE_BASE_NAME}:base-mitie-${BASE_MITIE_IMAGE_HASH} as mitie
 
 FROM ${IMAGE_BASE_NAME}:base-${BASE_IMAGE_HASH} as builder
 
+# install poetry
+ENV POETRY_VERSION ${POETRY_VERSION}
```
Comment (wochinge) [inline docker/Dockerfile.full:12]: Do you think having a separate `base-builder` image which extends the `base` image with the `poetry` installation is an overkill? Having that would reduce the duplicate steps for installing poetry.

**Review by wochinge — COMMENTED** [review]
💯

File: docker/Dockerfile.full (hunk 1 above)
Comment (tczekajlo) [reply docker/Dockerfile.full:12]: I like the idea!

**Review by wochinge — APPROVED** [review]
💯

**Pull request merged**